## TODO
* explain 'Helix' object
* document JSON file format.
* put in docstrings for all files
* show long-range stem pairs on 23S
* push middle layer to right
* complete 16S rRNA
* lay out other ribosome states.

### Future
* settings icon for toggles --> menu.
* automatic legend creation
* rename_tertiary_contact (for group II intron?)
* pathfinder routine for arrow linkers & noncanonical pairs
* K-turn assignment in rna_motif
* Read-in & auto-color motifs [SRL, etc.]
* convert to javascript
* Calculate 3D helix axis vectors in Rosetta and use to cross-check 2D helix orientations
* Allow DSSR setup of input files from PDB (skipping Rosetta)
* Initialization from secstruct alone (no PDB)
* Create a 'score' for total number of harmonious elements [e.g. lack of explicit linkers, mainly horizontal and vertical lines]
* For demo, produce 3-4 level 'game' 
* document MATLAB functions in .md, or use Sphinx --> move to RiboKit

