## TODO
* put in docstrings for helix
* put in docstrings for ligands
* put in docstrings for linkers
* put in docstrings for settings
* put in docstrings for tertiary_contacts

* document JSON file format.

* push 23S middle layer to right [spacing]
* show long-range stem pairs on 23S
* try fade out trick to recolor middle and back domain [and interconnecting arrows?].

* Write 'Getting Started' with tRNA & P4-P6 examples: explain 'Helix' object
* Write 'Multidomain RNAs documentation'
* complete 16S rRNA layout as test of tutorials and docs.
* Convert future to GitHub Issues.
 
### Future
* lay out other ribosome states.
* automatic legend creation (noncanonicals, stacks if visible, etc.)
* rename_tertiary_contact (for group II intron?)
* Read-in `.motifs.txt` & auto-color motifs [SRL, etc.]
* use open source SVG library to output drawings in SVG format.
* K-turn assignment in `rna_motif`
* Initialization from secstruct alone (no PDB)
* Allow DSSR setup of input files from PDB (skipping Rosetta)
* convert to javascript for web deployment and perhaps to allow faster i/o; MATLAB is slow.
* settings icon for toggles --> menu.
* completely automated pathfinder routine for arrow linkers & noncanonical pairs
* Calculate 3D helix axis vectors in Rosetta and use to cross-check 2D helix orientations
 
### Gamification
* Create a 'score' for total number of harmonious elements [e.g. lack of explicit linkers, mainly horizontal and vertical lines]
* animate rectangle flips/rotations
* show big bonuses when rectifying linkers.
* For demo, produce 3-4 level 'game' 

